src/pvrclient-mediaportal.cpp

```diff
@@ -2481,3 +2482,52 @@ PVR_ERROR cPVRClientMediaPortal::GetStreamReadChunkSize(int& chunksize)
   chunksize = 32 * 1024;
   return PVR_ERROR_NO_ERROR;
 }
+PVR_ERROR cPVRClientMediaPortal::GetRecordingEdl(const kodi::addon::PVRRecording& recording, std::vector<kodi::addon::PVREDLEntry>& edl) // bilkusg
+// liberally adapted from pvr.vuplus ( recordings.cpp in particular)
+{
+    cRecording* myrecording = GetRecordingInfo(recording);
+    const char* pvrFilePath = myrecording->FilePath();
+    const std::string& pvrSmbPath = m_tsreader->TranslatePath(pvrFilePath);
+    const std::string edlFile = pvrSmbPath.substr(0, pvrSmbPath.find_last_of('.')) + ".edl";
+    kodi::Log(ADDON_LOG_INFO, "GetRecordingEdl for %s from %s", pvrFilePath,edlFile.c_str());
+    FileReader *fileReader = new FileReader();
+    long fileOpenResult = fileReader->OpenFile(edlFile);
+    size_t fileSize = fileReader->GetFileSize();
+    unsigned char fileContentsChars[10000];//  = new unsigned char(fileSize + 1); // edl files aren't that large, so this is reasonable
+    size_t bytesRead = 0;
+    fileReader->Read(fileContentsChars, fileSize, &bytesRead);
+    fileReader->CloseFile();
+    SafeDelete(fileReader);
```

Comment:
If you use Kodi's file class this will be much cleaner and you don't need to use a large char buffer. You could instead have a helper function like:

https://github.com/kodi-pvr/pvr.iptvsimple/blob/4ac50e12742fd72a79fa4a7e1e79adbd17b8bf16/src/iptvsimple/utilities/FileUtils.cpp#L49-L61

Reply:
OK. I'll try to do that.